File: course-material/labs/extra/movie/solution/movie-database.cpp
#include "movie-database.h"


std::shared_ptr<Actor> MovieDatabase::create_actor(const std::string& name, const Date& birthday)
{
    auto actor = std::make_shared<Actor>(name, birthday);

    m_actors.push_back(actor);

    return actor;
}

std::shared_ptr<Movie> MovieDatabase::create_movie(const std::string& title)
{
    auto movie = std::make_shared<Movie>(title);

    m_movies.push_back(movie);

    return movie;
}

std::shared_ptr<Actor> MovieDatabase::find_actor(const std::string& name) const
{
    for (auto p : m_actors)
    {
        if (p->name() == name)
        {
            return p;
        }
    }

    return nullptr;
}

File: course-material/labs/extra/movie/java/MovieDatabase.java
import java.util.*;

class MovieDatabase
{
    private ArrayList<Actor> actors;
    private ArrayList<Movie> movies;

    public MovieDatabase()
    {
        this.actors = new ArrayList<>();
        this.movies = new ArrayList<>();
    }

    public Actor createActor(String name, Date birthday)
    {
        Actor actor = new Actor(name, birthday);
        actors.add(actor);
        return actor;
    }

    public Movie createMovie(String title)
    {
        Movie movie = new Movie(title);
        movies.add(movie);
        return movie;
    }

    public Actor findActor(String name)
    {
        for ( Actor actor : actors )
        {
            if ( name.equals(actor.name()) )
            {
                return actor;
            }
        }

        return null;
    }
}
    
